src/main/java/org/gridsuite/modification/server/modifications/NetworkStoreListener.java

```diff
@@ -275,4 +266,39 @@ private void flushEquipmentInfos() {
         equipmentInfosService.addAllEquipmentInfos(createdEquipments);
         equipmentInfosService.addAllEquipmentInfos(modifiedEquipments);
     }
+
+    private Set<AbstractBaseImpact> collectNetworkImpacts(Set<SimpleElementImpact> impacts) {
+        // keep Deletion impacts separatly
+        Set<AbstractBaseImpact> resImpacts = impacts.stream().filter(i -> i.getSimpleImpactType() == SimpleImpactType.DELETION).collect(Collectors.toSet());
+
+        // compute substations impact
+        if (impacts.stream().flatMap(i -> i.getSubstationIds().stream()).collect(Collectors.toSet()).size() >= collectionThreshold) {
+            resImpacts.add(CollectionElementImpact.builder()
+                    .elementType(IdentifiableType.SUBSTATION)
+                    .build());
+        }
+
+        // then filter those DELETION impacts for the next part and the collection impact computation
+        Set<SimpleElementImpact> filteredImpacts = impacts.stream()
+            .filter(i -> i.getSimpleImpactType() != SimpleImpactType.DELETION)
+            .collect(Collectors.toSet());
+        Set<IdentifiableType> impactTypes = filteredImpacts.stream()
+            .map(i -> i.getElementType()).collect(Collectors.toSet());
+
+        impactTypes.forEach(type -> {
+            // get SimpleElementimpacts of this equipment type
+            Set<SimpleElementImpact> typedNetworkImpacts = filteredImpacts.stream().filter(i -> type == i.getElementType()).collect(Collectors.toSet());
+            // compare using a threshold (ex: if nbLoadImpacts >= 50)
+            if (typedNetworkImpacts.size() >= collectionThreshold) {
+                resImpacts.add(CollectionElementImpact.builder()
+                    .elementType(type)
+                    .build());
+            } else {
+                // otherwise add the Simple Element impacts
+                resImpacts.addAll(typedNetworkImpacts);
+            }
+        });
```

Comment:
It would be cleaner to group impacts by type in a Map with Collectors.groupingBy, then do the rest of treatment.